### ISSUE TITLE
InitializationError on running JUnit runner

### ISSUE THREAD

**Issue by JeffGaly** [prashant-ramcharan/courgette-jvm#117]
Hi Prashant,

I am trying to configure courgette-jvm with JUnit to run parallel execution, I am facing InitializationError.Here is my build.gradle file dependency block , JUnit runner file and stack trace.Kindly help me to fix it.

build.gradle
-------------------
dependencies {
	 testCompile 'junit:junit:4.12'
     compile 'io.cucumber:cucumber-java8:4.3.0'
      compile 'io.cucumber:cucumber-java:4.3.0'
     compile 'io.cucumber:cucumber-core:4.3.0'
     compile 'io.cucumber:cucumber-junit:4.3.0'
     compile 'org.seleniumhq.selenium:selenium-java:3.141.59'
     testCompile 'org.hamcrest:hamcrest-all:1.3'   
     testCompile group: 'org.testng', name: 'testng', version: '6.8.21'
     testCompile 'io.cucumber:cucumber-jvm:4.3.0'
     compile 'io.github.prashant-ramcharan:courgette-jvm:3.1.0'
     compile fileTree(dir: 'libs', include: '*.jar')
}

RunnerFile.java
---------------------------
package ui.runners;

import org.junit.runner.RunWith;

import courgette.api.CourgetteOptions;
import courgette.api.junit.Courgette;
import cucumber.api.CucumberOptions;
import courgette.api.CourgetteRunLevel;

@RunWith(Courgette.class)
@CourgetteOptions(
        threads = 10,
        runLevel = CourgetteRunLevel.SCENARIO,
        rerunFailedScenarios = true,
        //rerunAttempts = 1,
        showTestOutput = true,
        //reportTargetDir = "build",
        cucumberOptions = @CucumberOptions(
                features = "src/test/resources/features/discover/",
                glue = "ui.steps",
                tags = {"@smoke"},
                plugin = {
                        "pretty",
                        "json:build/cucumber-report/cucumber.json",
                        "html:build/cucumber-report/cucumber.html",
                        "junit:build/cucumber-report/cucumber.xml"},
                strict = true
        ))

public class CourgetteTest {

}

StackTrace
----------------
java.lang.NoSuchMethodError: cucumber.api.CucumberOptions.format()[Ljava/lang/String;
	at courgette.runtime.CourgetteRuntimeOptions.createRuntimeOptions(CourgetteRuntimeOptions.java:87)
	at courgette.runtime.CourgetteRuntimeOptions.<init>(CourgetteRuntimeOptions.java:41)
	at courgette.runtime.CourgetteFeatureLoader.getCucumberFeatures(CourgetteFeatureLoader.java:27)
	at courgette.api.junit.Courgette.<init>(Courgette.java:39)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104)
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:33)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestLoader.createUnfilteredTest(JUnit4TestLoader.java:84)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestLoader.createTest(JUnit4TestLoader.java:70)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestLoader.loadTests(JUnit4TestLoader.java:43)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:444)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:678)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)

**Comment by prashant-ramcharan** [comment]
Hello,

Change your gradle deps to this:

````gradle
dependencies {
   testCompile 'junit:junit:4.12'
   compile 'org.seleniumhq.selenium:selenium-java:3.141.59'
   testCompile 'org.hamcrest:hamcrest-all:1.3'
   testCompile group: 'org.testng', name: 'testng', version: '6.8.21'
   compile 'io.github.prashant-ramcharan:courgette-jvm:3.3.0'
   compile fileTree(dir: 'libs', include: '*.jar')
}
````

Courgette will include the Cucumber deps for you so you don't need to specify the Cucumber deps in your build.gradle

**Comment by JeffGaly** [comment]
Hi Prashant ,
Thank you .I can execute my scenarios in parallel but im unable to see report properly 

![image](https://user-images.githubusercontent.com/32665795/59607111-4ff06100-90d8-11e9-815b-744dafbc402b.png)

though its running all features its showing only one feature as executed in report .Do we need to add any plugin for this ?

**Comment by prashant-ramcharan** [comment]
No, you don't need any additional plugins for the Courgette report. 

The Courgette report is generated using the Cucumber json report.

**Comment by JeffGaly** [comment]
what kind of plugin info i should give in runner so that it will fetch all 3 scenario in report.

**Comment by prashant-ramcharan** [comment]
Depends on how your features / scenarios are structured and tagged. I can see from you Courgette report, no Cucumber tags were provided.

If you can share your project / demo project, I can take a look but using the example project here would be a good starting point.

https://github.com/prashant-ramcharan/courgette-jvm-example

**Comment by JeffGaly** [comment]
Hi Prashant,

I gave tags to feature as @ smoke but still it shows single feature report indtead of 3 .

![image](https://user-images.githubusercontent.com/32665795/59742498-75de4880-9233-11e9-9a2d-39dcc34f7cbd.png)


My Runner


@RunWith(Courgette.class)
@CourgetteOptions(
        threads = 4,
        runLevel = CourgetteRunLevel.FEATURE,
        rerunFailedScenarios = false,
        rerunAttempts = 1,
        showTestOutput = true,
        reportTargetDir = "target",
        cucumberOptions = @CucumberOptions(
                features = "src/test/resources/features/discover/",
                glue = "ui.steps",
                tags = {"@smoke"},
                plugin = {
                                "pretty",
                                "json:target/cucumber-report/cucumber.json",
                                "html:target/cucumber-report/cucumber.html",
                                "junit:target/cucumber-report/cucumber.xml"},
                	strict = true
        ))

public class courgetteJunit {

}

**Comment by prashant-ramcharan** [comment]
Please create a new issue and provide a project that I can reproduce as I'm unable to advise any further without seeing your project / structure. 

Thanks.
